pkg/sca/apptypecheck.go

```diff
@@ -0,0 +1,107 @@
+package sca
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/salsadigitalauorg/shipshape/pkg/shipshape"
+	"github.com/salsadigitalauorg/shipshape/pkg/utils"
+)
+
+const AppType shipshape.CheckType = "sca:application_type"
+
+/*
+Example:
+	threshold: 20
+	disallowed:
+		- symfyony
+	markers:
+		symfony:
+			- "    return new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']);"
+	dirs:
+		symfony:
+			- bin
+			- config
+			- public
+	dependencies:
+		symfony:
+			- symfony/runtime
+			- symfony/symfony
+			- symfony/framework
+
+@TODO: currently only supports composer for dependency lookups.
+*/
+type AppTypeCheck struct {
+	shipshape.CheckBase `yaml:",inline"`
+	Threshold           int                 `yaml:"threshold"`
+	Disallowed          []string            `yaml:"disallowed"`
+	Entrypoint          string              `yaml:"entrypoint"`
+	Paths               []string            `yaml:"paths"`
+	Markers             map[string][]string `yaml:"markers"`
+	Dirs                map[string][]string `yaml:"dirs"`
+	Dependencies        map[string][]string `yaml:"dependencies"`
+}
+
+func (c *AppTypeCheck) Merge(mergeCheck shipshape.Check) error {
+	appTypeCheck := mergeCheck.(*AppTypeCheck)
+	if err := c.CheckBase.Merge(&appTypeCheck.CheckBase); err != nil {
+		return err
+	}
+
+	utils.MergeStringSlice(&c.Disallowed, appTypeCheck.Disallowed)
+	utils.MergeStringSlice(&c.Paths, appTypeCheck.Paths)
+	c.Threshold = appTypeCheck.Threshold
+
```

Comment:
Mostly because I wasn't sure how the merge would happen with `map[string][]string` as opposed to `[]string`, but also because I thought we wouldn't be changing the markers when we override.